modules/custom/az_migration/src/Plugin/migrate/process/ParagraphsBehavior.php

```diff
@@ -20,7 +20,35 @@ class ParagraphsBehavior extends ProcessPluginBase {
    */
   public function transform($value, MigrateExecutableInterface $migrate_executable, Row $row, $destination_property) {
     // Setting the behavior to the paragraph.
-    $value['behavior'] = serialize(['az_display_settings' => ['bottom_spacing' => $row->getSourceProperty('bottom_spacing')]]);
+    $behavior = ['az_display_settings' => ['bottom_spacing' => $row->getSourceProperty('bottom_spacing')]];
+    if (!empty($this->configuration['gallery_display'])) {
+      $behavior['gallery_display'] = $this->configuration['gallery_display'];
+    }
+    if (!empty($this->configuration['bg_color'])) {
+      $bg_color = $row->getSourceProperty($this->configuration['bg_color']);
+      $behavior['az_text_background_paragraph_behavior']['text_background_color'] = '';
+      foreach ($bg_color as $bg_color_item) {
+        $behavior['az_text_background_paragraph_behavior']['text_background_color'] .= $bg_color_item['value'];
+      }
+    }
+    if (!empty($this->configuration['bg_pattern'])) {
+      $bg_pattern = $row->getSourceProperty($this->configuration['bg_pattern']);
+      $behavior['az_text_background_paragraph_behavior']['text_background_pattern'] = '';
+      foreach ($bg_pattern as $bg_pattern_item) {
+        $match_pattern_item = [
+          'bg-triangles-mosaic',
+          'bg-triangles-fade',
+          'bg-catalinas-abstract',
+        ];
+        if (in_array($bg_pattern_item['value'], $match_pattern_item)) {
+          $behavior['az_text_background_paragraph_behavior']['text_background_pattern'] .= 'None';
+        }
+        else {
+          $behavior['az_text_background_paragraph_behavior']['text_background_pattern'] .= $bg_pattern_item['value'];
```

Comment:
The behavior dropdown for this only allows one choice, so avoid the `.=` concatenate operator. If multiple choices were somehow in the migration, the concatenated strings would probably not match any of the options.

Reply:
Done